fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/api/IIdNameCache.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.dao.api;
+
+import java.util.Map;
+
+/**
+ * Interface to a cache mapping an id of type T to a string. Supports
+ * thread-local caching to support temporary staging of values pending
+ * successful completion of a transaction.
+ * @param<T>
+ */
+public interface IIdNameCache<T> {
```

Comment:
I think you're just following the pattern of the INameIdCache but I don't really understand why we need interfaces for these...will they ever have more than one implementation?

Reply:
Yeah, just wanted to be consistent with the other caches.

Reply:
I like to hide implementation details where possible so we can change the underlying implementation in the future if we need to.